packages/node/src/indexer/dictionary/ethDictionary.service.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import { Inject, Injectable } from '@nestjs/common';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { NETWORK_FAMILY } from '@subql/common';
+import { NodeConfig, DictionaryService } from '@subql/node-core';
+import { EthereumBlock, SubqlDatasource } from '@subql/types-ethereum';
+import { SubqueryProject } from '../../configure/SubqueryProject';
+import { EthDictionaryV1 } from './v1';
+import { EthDictionaryV2 } from './v2';
+
+@Injectable()
+export class EthDictionaryService extends DictionaryService<
+  SubqlDatasource,
+  EthereumBlock
+> {
+  async initDictionaries(): Promise<void> {
+    let dictionaryV1Endpoints: string[] = [];
+    const dictionariesV2: EthDictionaryV2[] = [];
+
+    if (!this.project) {
+      throw new Error(`Project in Dictionary service not initialized `);
+    }
+    const registryDictionaries = await this.resolveDictionary(
+      NETWORK_FAMILY.substrate,
+      this.project.network.chainId,
+      this.nodeConfig.dictionaryRegistry,
+    );
+
+    const dictionaryEndpoints: string[] = (
+      !Array.isArray(this.project.network.dictionary)
+        ? !this.project.network.dictionary
+          ? []
+          : [this.project.network.dictionary]
+        : this.project.network.dictionary
+    ).concat(registryDictionaries);
```

Comment:
This is duplicated with the substrate code. I think we should be able to move it to node core.